worldengine/draw.py

```diff
@@ -406,6 +406,113 @@ def draw_biome(world, target):
             v = biome[y, x]
             target.set_pixel(x, y, _biome_colors[v])
 
+def draw_scatter_plot(world, size, target):
+    """ This function can be used on a generic canvas (either an image to save
+        on disk or a canvas part of a GUI)
+    """
+    min_humidity = None
+    max_humidity = None
+    min_temperature = None
+    max_temperature = None
+    for y in range(world.height):
```

Comment:
```python
    min_humidity = world.humidity['data'].min()
    max_humidity = world.humidity['data'].max()
    min_temperature = world.temperature['data'].min()
    max_temperature = world.temperature['data'].min()
```
This can replace the loops between line 417 and 429. (And, of course, lines 413 to 416.)